lib/elixir/lib/dynamic_supervisor.ex

```diff
@@ -0,0 +1,859 @@
+defmodule DynamicSupervisor do
+  @moduledoc ~S"""
+  A supervisor that starts children dynamically.
+
+  The `Supervisor` module was designed to handle mostly static children
+  that are started in the given order when the supervisor starts. A
+  `DynamicSupervisor` starts with no children. Instead, children are
+  started on demand via `start_child/2`. When a dynamic supervisor
+  terminates, all children are shutdown at the same time, with no guarantee
+  of ordering.
+
+  ## Examples
+
+  A dynamic supervisor is started with no children, only with the
+  supervision strategy (the only strategy currently supported is
+  `:one_for_one`):
+
+      {:ok, sup} = DynamicSupervisor.start_link(strategy: :one_for_one)
+
+  Once the dynamic supervisor is running, we can start children
+  with `start_child/2`, which receives a child specification:
+
+      {:ok, agent1} = DynamicSupervisor.start_link(sup, {Agent, fn -> %{} end})
+      Agent.update(agent1, &Map.put(&1, :key, "value"))
+      Agent.get(agent1, & &1)
+      #=> %{key: "value"}
+
+      {:ok, agent2} = DynamicSupervisor.start_link(sup, {Agent, fn -> %{} end})
+      Agent.get(agent2, & &1)
+      #=> %{}
+
+      DynamicSupervisor.count_children(sup_pid)
+      #=> %{active: 2, specs: 2, supervisors: 0, workers: 2}
+
+  ## Module-based supervisors
+
+  Similar to `Supervisor`, dynamic supervisors also support module-based
+  supervisors.
+
+      defmodule MyApp.DynamicSupervisor do
+        # Automatically defines child_spec/1
+        use DynamicSupervisor
+
+        def start_link(arg) do
+          DynamicSupervisor.start_link(__MODULE__, arg, name: __MODULE__)
+        end
+
+        def init(_arg) do
+          DynamicSupervisor.init(strategy: :one_for_one)
+        end
+      end
+
+  See the `Supervisor` docs for a discussion of when you may want to use
+  module-based supervisors.
+
+  ## Name registration
+
+  A supervisor is bound to the same name registration rules as a `GenServer`.
+  Read more about these rules in the documentation for `GenServer`.
+  """
+
+  @behaviour GenServer
+
+  @doc """
+  Callback invoked to start the supervisor and during hot code upgrades.
+  """
+  @callback init(args :: term) :: {:ok, sup_flags()} | :ignore
+
+  @typep sup_flags() :: %{
```

Comment:
I'm confused with `intensity` and `period`. Are these renamed `max_restarts` and `max_seconds`?